packages/components/src/dropdown-menu/style.scss

```diff
@@ -30,19 +30,30 @@
 			height: 1px;
 		}
 
-		&.is-active svg {
-			// Block UI appearance.
-			color: $white;
-			background: $gray-900;
-			box-shadow: 0 0 0 $border-width $gray-900;
-			border-radius: $border-width;
+		&.is-active {
+			svg,
+			span.dashicon {
+				// Block UI appearance.
+				color: $white;
+				background: $gray-900;
+				box-shadow: 0 0 0 $border-width $gray-900;
+				border-radius: $border-width;
+			}
 		}
 
 		// Formatting buttons
-		> svg {
+		> svg,
+		> span.dashicon {
 			border-radius: $radius-block-ui;
 			width: $button-size-small;
 			height: $button-size-small;
+			margin: 0 8px 0 0;
+		}
+
+		> span.dashicon {
+			display: flex;
+			align-items: center;
+			justify-content: center;
```

Comment:
Just checking — have we considered whether these margin and flex styles should be in the Button component itself? I couldn't really check because of #43700, but it seems like that might be the case.

Reply:
Thanks for digging!

I don't have an opinion on whether dashicons in Button should be 20 or 24px. If we do want it to be 24px, we can do #43724. If we want to keep it at 20px, I think we should wrap each Button [Icon](https://github.com/WordPress/gutenberg/blob/132b86cd48e4f785a249175476302c66dc95ed2c/packages/components/src/button/index.js#L160-L166) in a centering flex container at `iconSize` width/height so we can maintain a consistent size between dashicons and SVGs. That would be more robust than the hardcoded 2px margin thing that is going on right now.

I also made an issue (#43725) for the deprecation suggested in the PR you mentioned. Though, I imagine we would need a long deprecation period, because it's probably very widely used.

Reply:
I have been rethinking what the best way to go about this is.

I think the size of the icon in the `Button` component should be only defined by the `iconSize` prop and not explicitly given width and height.
In other words, dashicons should remain at the default 20px, or 24px for SVG icons, and only when the `iconSize` prop is specified should the size of the element (font-size for dashicons) change according to that value.
(However, as noted in #43724, the iconSize property does not currently support dashicons.)

Therefore, it would make sense that when a dashicon is used in a `Button` component, it is adjusted to the default 20px size. However, in `DropdownMenu`, the size of the background color that represents the active state must be consistent with the size of the element, whether it is an svg icon or a dashicon.

Therefore, I have finally decided that this change by PR makes sense, what do you think?

Reply:
I'm only skeptical about the margin and flex styles being here (lines 50–56). The margin override is especially hacky. I'm thinking it could be cleaner if we cleaned it up upstream so it doesn't have the hacky side margins to begin with:

https://github.com/WordPress/gutenberg/blob/e935c5d0ac1e206362307ff22f5318de42f0d822/packages/components/src/button/style.scss#L309-L314

↓

```scss
		.dashicon {
			display: inline-flex;
			justify-content: center;
			align-items: center;
			flex: 0 0 auto;
		}
```

(I haven't tested in all Button variants, but something like that.)

Reply:
Thanks for the advice!
I was looking for a fix in the upstream button component and found two issues:

- #44043
- #44042

I would like to put these two issues as blockers and hold off on moving forward with this PR, as I think these two should be resolved first.